osu.Game/Screens/Ranking/SoloResultsScreen.cs

```diff
@@ -1,49 +1,40 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-using System;
-using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
-using osu.Framework.Allocation;
-using osu.Game.Beatmaps;
-using osu.Game.Extensions;
-using osu.Game.Online.API;
-using osu.Game.Online.API.Requests;
-using osu.Game.Rulesets;
+using osu.Framework.Bindables;
 using osu.Game.Scoring;
 
 namespace osu.Game.Screens.Ranking
 {
     public partial class SoloResultsScreen : ResultsScreen
     {
-        private GetScoresRequest? getScoreRequest;
+        public readonly IBindableList<ScoreInfo> Scores = new BindableList<ScoreInfo>();
```

Comment:
So looking at the actual code, my primary question here would be - does this _need_ to be bindable?

Once you enter results, the scores are not going to change until you exit back to song select. So is the complexity incurred from handling having this change worth it?

Additionally, there's also https://github.com/ppy/osu/issues/27609 which is still an issue even with this bindable, the score retrieval still strongly depends on song select (which it shouldn't).

Dunno. All this is making me question whether further review of this diff is worth the time.

Reply:
The only difference is that if the results are delayed, then the results wouldn't show. One of the tests tested delayed fetching, so I didn't wanted to break that. This is highly unluckily and I wrote my concerns about it in my initial comment.

I also agree that the whole score retrival should be refactored. I can do that but you probably don't want me to. I decided to take scores in the same fashion as `SoloGameplayLeaderboard`, so the eventual refactoring might be easier.